composer/src/main/kotlin/com/gojuno/composer/TestRun.kt

```diff
@@ -173,7 +173,7 @@ private fun pullTestFiles(adbDevice: AdbDevice, test: InstrumentationTest, outpu
             PulledFiles(
                     files = emptyList(), // TODO: Pull test files.
                     screenshots = screenshotsFolderOnHostMachine.let {
-                        when (it.exists()) {
+                        when (it.exists() && it.listFiles() != null) {
```

Comment:
`listFiles()` can return `null` only if `it` is not a directory. Calling `exists()` beforehand is partially redundant IMHO. 
There is also a race condition possible when directory is removed in the time between two `listFiles()` calls.

What about replacing `when` block with `it.listFiles()?.toList() ?: emptyList()` ?

Reply:
What about `when (it.exists() && it.isDirectory())`?

Race condition is still possible with `it.listFiles()?.toList() ?: emptyList()` since we're going to get problems later during `adb pull` if list of files is invalid

Reply:
`isDirectory()` returns `false` if there is no directory under given path including the case when there is nothing there.

> we're going to get problems later during adb pull if list of files is invalid

It seems that screenshots have been already pulled when we are listing those files:
https://github.com/rafakob/composer/blob/f70e773c73adc6e7f98ffaf85f0000d5f8675ee4/composer/src/main/kotlin/com/gojuno/composer/TestRun.kt#L164

Reply:
ah right, my bad

>returns false if there is no directory under given path including the case when there is nothing there.

Isn't that what we want?

Reply:
Yes, but `listFiles() != null` also checks for existence.

Reply:
Well for sake of removing redundancy from if statement I'm 👍 with using `listFiles() != null`